docs/sources/reference/components/loki/loki.source.file.md

```diff
@@ -40,13 +40,17 @@ The component starts a new reader for each of the given `targets` and fans out l
 | `encoding`              | `string`             | The encoding to convert from when reading files.                            | `""`    | no       |
 | `tail_from_end`         | `bool`               | Whether a log file is tailed from the end if a stored position isn't found. | `false` | no       |
 | `legacy_positions_file` | `string`             | Allows conversion from legacy positions file.                               | `""`    | no       |
+| `retry_interval`        | `duration`           | Frequency to try re-opening files that were closed.                         | `"0s"`  | no       |
 
 The `encoding` argument must be a valid [IANA encoding][] name. If not set, it
 defaults to UTF-8.
 
 You can use the `tail_from_end` argument when you want to tail a large file without reading its entire content.
 When set to true, only new logs will be read, ignoring the existing ones.
 
+`retry_interval` is deactivated by default (`"0s"`). This should be set on Windows-based systems when rotating files with the same names because the
+component will no try to re-open the files and the targets won't be updated because of the cache.
+This is not needed on Unix-like systems because the component will always try to re-open the deleted files.
```

Comment:
```suggestion
This argument is not required on Linux, macOS, or FreeBSDs because`loki.source.file` always tries to re-open the closed files.
```

We can state the specific OSes we support.

This sentence (and the overall description) isn't really clear to me. Why will `loki.source_file` try to reopen **deleted** files? The description in the table says the arg will tell the component to try to reopen **closed** files.  Which is it?  I assume it's closed files since trying to reopen deleted files doesn't really make sense.  Why won't the component try to reopen closed files on Windows? It's also not really clear to me. Windows caches the files? And  the other OSes do not?

Reply:
We have a case where a user is rotating a "app.log" file everytime it reaches 5MB by deleting it and creating a new app.log file.

On Windows you must first close the file before deleting it. Closing the file will stop the tailer in the loki.source.file component. If the new file would have a different name, the local.file_match would send it to the loki.source.file component and it would work fine. But for the local.file_match component nothing changed because it only checks at a regular interval and it will just see that there is still an "app.log" file. Because of this, it won't notify the loki.source.file component to start a new tailer and the file won't be tailed (it only notifies the next component when the set of files changed). 

On Unix-like systems, you can delete the file without closing it. Because the file is not closed, the tailer is not stopped, it continuously tries to read the file until the local.file_match sends an update. In the case of the user, it will restart the tailer once the new "app.log" file is created because of the names it thinks that it's the same file.